build_tools/cmake/iree_copts.cmake

```
@@ -306,14 +306,29 @@ if(ANDROID)
   )
 endif()
 
+if (NOT ${CMAKE_SYSTEM_NAME} STREQUAL "Generic")
```

This could use a comment, or a more specific check.

```
// If building for a known OS, link against libdl for dynamic library support.
```

```
// Generic systems may not support dynamic libraries.
```

also nit: no space
```suggestion
if(NOT ${CMAKE_SYSTEM_NAME} STREQUAL "Generic")
```

---

Done.